### PR TITLE
update to bevy 0.14

### PR DESCRIPTION
**Pull request by JackCrumpLeys**
update to bevy 0.14. works towards #620  task 3.

### CODE REVIEW

**Comment by JackCrumpLeys** [comment]
I did some crazy git stuff. oops fixing it now

**Comment by JackCrumpLeys** [comment]
Ok I think I solved it

**Comment by JackCrumpLeys** [comment]
Can this be reviewed @dyc3 @rj00a? It is not a huge pr. Thanks.

File: crates/valence_command/src/manager.rs [hunk 1]
```diff
@@ -151,10 +151,11 @@ fn update_client_command_tree(
         let mut new_root = None;
 
         while let Some((parent, node)) = to_visit.pop() {
-            if already_visited.contains(&(parent.map(|(_, edge)| edge), node)) {
+            if already_visited.contains(&(parent.map(|(node_id, _)| node_id), node)) {
                 continue;
             }
-            already_visited.insert((parent.map(|(_, edge)| edge), node));
+            already_visited.insert((parent.map(|(node_id, _)| node_id), node));
+
```
Comment (dyc3) [inline crates/valence_command/src/manager.rs:158]: This change seems unrelated to the upgrade.

File: crates/valence_server/Cargo.toml [hunk 2]
```diff
@@ -8,9 +8,6 @@ repository.workspace = true
 documentation.workspace = true
 license.workspace = true
 
-[lints]
-workspace = true
-
```
[the remark this answers is not in the document]
Comment (JackCrumpLeys) [reply crates/valence_server/Cargo.toml:13]: This is part of the git issues I was having and it slipped though my hands. I accidentally commented the main branch from 6 mouths ago and had to prune all the changes manually.

File: crates/valence_command/src/manager.rs (hunk 1 above)
Comment (JackCrumpLeys) [reply crates/valence_command/src/manager.rs:158]: I will make a new pr for this

File: crates/valence_server/src/status.rs [hunk 3]
```diff
@@ -32,12 +32,16 @@ fn handle_status(
...
+                ClientStatusC2s::PerformRespawn => {
+                    respawn_events.send(RequestRespawnEvent {
+                        client: packet.client,
+                    });
+                }
+                ClientStatusC2s::RequestStats => {
+                    request_stats_events.send(RequestStatsEvent {
+                        client: packet.client,
+                    });
+                }
```
[the remark this answers is not in the document]
Comment (JackCrumpLeys) [reply crates/valence_server/src/status.rs:44]: In this case the event type now returns something (an eventid) looking back I should have just put a semicolon at the end there

[the remark this answers is not in the document]
Comment (JackCrumpLeys) [reply crates/valence_server/src/status.rs:44]: disregard I forgot that every event id has a different type so it cannot be done this way

**Comment by dyc3** [comment]
I'll review it when CI is passing.

**Comment by JackCrumpLeys** [comment]
> I'll review it when CI is passing.
ci is passing .

**Comment by JackCrumpLeys** [comment]
> I'll review it when CI is passing.

@dyc3 CI is passing

**Review by dyc3 — APPROVED** [review]
